### ISSUE TITLE
De-Class Arduino Code

### ISSUE THREAD

**Issue by BigBroccoli** [SIU-Robotics/moondawg-ino#42]
Get rid of all the classes in the Arduino code and make all of the files "Helper Function Files". This will be the new standard moving forward

**Comment by BigBroccoli** [comment]
Was having multiple issues with functions having multiple definitions
fixed by: 
declaring the ``` Servo ``` objects as extern in the header files and define them in the ``` .cpp ``` file ex:
.h - ``` extern Servo digMotor; ```
.cpp - ``` Servo digMotor; ```

According to ChatGPT the issue is

> The issue you're facing arises because you are defining the Servo objects (digMotor, actuator, auger, etc.) in the header file. When you include this header file in multiple source files, the compiler ends up creating multiple definitions of these objects, causing the "multiple definition" error.

**Comment by BigBroccoli** [comment]
For more info on extern

<details>
<summary>ChatGPT</summary>
<br>
In C++, the `extern` keyword is used primarily in two contexts:

### 1. **Declaring Variables Across Multiple Files (External Linkage)**
   The `extern` keyword is often used to declare a variable that is defined in another file or translation unit. This allows different source files to share the same global variable.

   - **Without `extern`:** If you define a global variable in one file, it is only visible within that file.
   - **With `extern`:** You declare the variable in one file using `extern` and define it in another file, making it accessible across files.

   **Example**:
   
   ```cpp
   // File1.cpp
   int x = 10; // Definition

   // File2.cpp
   extern int x; // Declaration (no memory is allocated here)
   void printX() {
       std::cout << x << std::endl; // Uses the variable from File1.cpp
   }
   ```

### 2. **Declaring Functions (Optional)**
   In C++, functions have external linkage by default, so using `extern` with functions is not strictly necessary. However, it can still be used to explicitly declare that a function has external linkage.

   **Example**:
   
   ```cpp
   extern void myFunction(); // Declaration of a function defined elsewhere
   ```

### 3. **`extern "C"` for C Compatibility**
   C++ name mangling can cause problems when linking C and C++ code. To prevent this, the `extern "C"` keyword is used to tell the C++ compiler not to apply name mangling to the enclosed code so that it can be linked with C code.

   **Example**:
   
   ```cpp
   extern "C" {
       void myCFunction(); // This will have C linkage, not C++ linkage
   }
   ```

In summary, `extern` is used to declare variables or functions that are defined elsewhere, typically in a different file or language, facilitating cross-file or cross-language linkage.
</details>


Other sources:
[Stackoverflow](https://stackoverflow.com/a/10422050)
